packages/snap-preact/src/Snap.tsx

```diff
@@ -363,7 +363,7 @@ export class Snap {
 			isMergeableObject: isPlainObject,
 		});
 
-		if (!this.config?.client?.globals?.siteId && this.context.siteId) {
+		if (this.context.siteId) {
```

Comment:
🤔 ... could this change possibly cause problems to existing sites that are providing both the client global and the siteId in the context?